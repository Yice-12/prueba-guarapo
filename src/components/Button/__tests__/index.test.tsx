import { fireEvent, render } from "@testing-library/react";
import { Button } from "../";

test("render button and click", () => {
  const mockSetPage = jest.fn();

  const { asFragment, getByText } = render(
    <Button page={1} setPage={mockSetPage} />
  );

  expect(asFragment()).toMatchSnapshot();

  const button = getByText("LOAD MORE");
  fireEvent.click(button);

  expect(mockSetPage).toHaveBeenCalledTimes(1);
});
