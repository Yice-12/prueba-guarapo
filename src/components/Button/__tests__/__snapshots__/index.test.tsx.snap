// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button and click 1`] = `
<DocumentFragment>
  <div
    class="button"
  >
    <div
      class="button__content"
    >
      <button>
        LOAD MORE
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders button text 1`] = `
<DocumentFragment>
  <div
    class="button"
  >
    <div
      class="button__content"
    >
      <button>
        LOAD MORE
      </button>
    </div>
  </div>
</DocumentFragment>
`;
